streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedTasks.java

```diff
@@ -434,14 +423,30 @@ void close(final boolean clean) {
     }
 
     private void close(final Collection<Task> tasks, final boolean clean) {
-        for (final Task task : tasks) {
+        for (Iterator<Task> it = tasks.iterator(); it.hasNext(); ) {
```

Comment:
Don't think we need to change it from the for each loop. We don't need to remove the closed tasks as they are immediately cleared after the call to this method

Reply:
Only because we call `close()` in a single place only -- does not to be future prove. Btw: this raises the question, why we need this `private` method -- doesn't add much IMHO. Can we just remove it and put the code in `void close(final boolean clean)` ?

Reply:
yeah, i was thinking we could inline this method into `close`, too